tests/unit_tests/test_suite_test.go

```diff
@@ -0,0 +1,91 @@
+package tests
+
+import (
+	"bytes"
+	"io"
+	"os"
+	"testing"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"kubevirt.io/client-go/kubecli"
+
+	instancetypev1beta1 "kubevirt.io/api/instancetype/v1beta1"
+)
+
+const (
+	testNamespace                  = "common-instancetype-functest"
+	clusterInstanceTypesBundlePath = "../../common-clusterinstancetypes-bundle.yaml"
+	clusterPreferencesBundlePath   = "../../common-clusterpreferences-bundle.yaml"
+)
+
+var (
+	virtClient kubecli.KubevirtClient
+
+	virtualMachineClusterInstancetypes []instancetypev1beta1.VirtualMachineClusterInstancetype
+	virtualMachineClusterPreferences   []instancetypev1beta1.VirtualMachineClusterPreference
+	err                                error
+)
+
+type clusterType interface {
+	instancetypev1beta1.VirtualMachineClusterInstancetype | instancetypev1beta1.VirtualMachineClusterPreference
+}
+
+func decodeResources[C clusterType](b []byte) ([]C, error) {
```

Comment:
Can we import this from somewhere? IIRC this is copied from `kubevirt/kubevirt`? If so please keep it and add a comment? We don't want to import `kubevirt/kubevirt`.

Reply:
It is not from `kubevirt/kubevirt`, but I took this code from the SSP operator. I think it is better to add it like this instead of importing since it is just a small piece of code that is needed. I will add a comment that it is taken from SSP.

Reply:
Ok, please add a note so we can see all copied funcs with a reference to the soruce. You could surround them with a comment block to make it more visual.

Reply:
I meant similar to like

```
// following funcs taken from ssp
func1
func2
func3
func3
// end of funcs taken from ssp
```